### PR TITLE
pyflakes fixes and CI

### PR DESCRIPTION
**Pull request by martinpitt**

### CODE REVIEW

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

File: example.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ async def main():
                                             's', 'org.freedesktop.hostname1')
     print(items)
 
-    slot = system.add_match("interface='org.freedesktop.DBus.Properties'", property_changed)
+    system.add_match("interface='org.freedesktop.DBus.Properties'", property_changed)
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply example.py:46]: Sure, the return value is discarded, but the actual match/callback work without, I tested that. Presumably the reference to `property_changed` is stored internally, which keeps the slot alive. Anyway, I added a `del`.

File: systemd_ctypes/__init__.py [hunk 2]
```diff
@@ -22,3 +22,5 @@
 from .bus import Bus, BusError, BusMessage
 from .event import Event, EventLoopPolicy
 from .pathwatch import PathWatch
+
+__all__ = [ Bus, BusError, BusMessage, Event, EventLoopPolicy, PathWatch ]
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply systemd_ctypes/__init__.py:26]: That's what I started with, then ale reformatted it like this and I shrugged.. :grin: Done.

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -26,7 +29,7 @@ jobs:
         run: |
           if type apt >/dev/null 2>&1; then
               apt-get update
-              DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends python3-dbusmock libsystemd0
+              DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends python3-dbusmock libsystemd0 ${{ matrix.pyflakes }}
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply .github/workflows/tests.yaml:32]: It's extra wasted time due to the download and install, but if you prefer, I can install it on Debian as well. Done.

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -37,6 +40,14 @@ jobs:
       - name: Run tests
         run: test/run
 
+      - name: pyflakes
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply .github/workflows/tests.yaml:43]: Heh, when I wrote this PR, tox was not in main yet. I was aware that you were going to drop test/run, which left me without a main entry point. Scanning installed files sounds wrong for static code checks -- if for no other reason that it breaks the file paths.

Agreed that there should be some single hook, I am just not sure where you would want to have it. In dbusmock I have [tests/test_code.py](https://github.com/martinpitt/python-dbusmock/blob/master/tests/test_code.py), i.e. "just another test". If you want that, I'm happy to do it that way.

[the remark this answers is not in the document]
Comment (allisonkarlitskaya) [reply .github/workflows/tests.yaml:43]: I was thinking about that: it's actually a cool approach.

Would you invoke it by `import pyflake`?  Or `subprocess`?

[the remark this answers is not in the document]
Comment (martinpitt) [reply .github/workflows/tests.yaml:43]: I didn't try running it via `import pyflakes`; so far I only ever ran it through `sys.interpreter -m pyflakes`, which works, and I didn't care enough to spend more time on that (I'm much more of a "good enough" developer wrt. build systems; I know you  are a lot more curiuos and a perfectionist there :grin: ).

**Comment by allisonkarlitskaya** [comment]
I'm gonna drop the second commit from #14, land it, and then we can rebase this one on that and add the checks to the `tox` invocation.

**Comment by martinpitt** [comment]
This needs to be re-done completely, and @allisonkarlitskaya said she wanted to look at that now. It's also much easier now with the tox runner, it can just be added there.

**Pull request closed without merging**